### ISSUE TITLE
Set Implicit Arguments seems interfere with K

### ISSUE THREAD

**Issue by HuStmpHrrr** [mattam82/Coq-Equations#172]
Following program gives an error
```
Error: Anomaly "Uncaught exception Not_found."
Please report at http://coq.inria.fr/bugs/.
```
The program is
```coq
Set Implicit Arguments.

From Equations Require Export Equations.
Require Export Equations.Subterm.

Require Import Equations.DepElimK.

(* Module KAxiom. *)

(** By default we disallow the K axiom, but it can be set. *)

Set Equations WithK.

(** In this case the following definition uses the [K] axiom just imported. *)

Equations K {A} (x : A) (P : x = x -> Type) (p : P eq_refl)
          (H : x = x) : P H :=
  K x P p eq_refl := p.
```

However, if the `Set Implicit Arguments` is commented out, then `K` can be defined.

**Comment by mattam82** [comment]
Interresting, I’ll investigate

**Comment by mattam82** [comment]
Fixed in master, will be in the final 1.2.
